include/deal.II/base/template_constraints.h

```diff
@@ -466,6 +466,17 @@ struct EnableIfScalar<std::complex<T>>
 };
 
 
+
+template <typename U, typename... T>
+constexpr bool contains(std::tuple<T...>)
+{
+  return !std::is_same<
+    std::integer_sequence<bool, false, std::is_same<U, T>::value...>,
+    std::integer_sequence<bool, std::is_same<U, T>::value..., false>>::value;
+}
```

Comment:
Would you mind adding some Doxygen documentation for this function?

Reply:
Sure, is c1d49e7 OK?

I'm recompiling the code in order to check that I did not break anything.